extensions/skins/conversation_v1/Conversation.js

```diff
@@ -75,227 +218,425 @@ oppia.directive('conversationSkin', [function() {
+              oppiaPlayerService.getInteractionHtml(
+                newStateName, _nextFocusLabel
+              ) + oppiaPlayerService.getRandomSuffix() : '');
+
+            if (feedbackHtml) {
+              lastAnswerFeedbackPair.oppiaFeedback = feedbackHtml;
+              $scope.waitingForContinueButtonClick = true;
+              focusService.setFocus($scope.CONTINUE_BUTTON_FOCUS_LABEL);
+              scrollToBottom();
             } else {
-              if (feedbackHtml) {
-                $scope.waitingForOppiaFeedback = false;
-                $scope.waitingForNewCard = true;
-                _scrollToBottom(function() {
-                  $timeout(function() {
-                    $scope.waitingForNewCard = false;
-                    _addNewCard($scope.stateName, questionHtml);
-                    _scrollToBottom(function() {
-                      if (_learnerInputIsInView) {
-                        focusService.setFocus(_labelForNextFocusTarget);
-                      }
-                      _answerIsBeingProcessed = false;
-                    });
-                  }, 1000);
+              // Note that feedbackHtml is an empty string if no feedback has
+              // been specified. This causes the answer-feedback pair to change
+              // abruptly, so we make the change only after the animation has
+              // completed.
+              $scope.showPendingCard(
+                newStateName,
+                contentHtml + oppiaPlayerService.getRandomSuffix(),
+                function() {
+                  lastAnswerFeedbackPair.oppiaFeedback = feedbackHtml;
                 });
-              } else {
-                $scope.waitingForOppiaFeedback = false;
-                _addNewCard($scope.stateName, questionHtml);
-                _scrollToBottom(function() {
-                  if (_learnerInputIsInView) {
-                    focusService.setFocus(_labelForNextFocusTarget);
-                  }
-                  _answerIsBeingProcessed = false;
-                });
-              }
             }
-          }, millisecsLeftToWait);
+          }
+
+          $scope.finished = oppiaPlayerService.isStateTerminal(newStateName);
+          _answerIsBeingProcessed = false;
+        }, true);
+      };
+
+      $scope.startCardChangeAnimation = false;
+      $scope.showPendingCard = function(newStateName, newContentHtml, successCallback) {
+        $scope.waitingForContinueButtonClick = false;
+        $scope.startCardChangeAnimation = true;
+
+        $timeout(function() {
+          _addNewCard(
+            newStateName,
+            newContentHtml,
+            oppiaPlayerService.getInteractionHtml(
+              newStateName, _nextFocusLabel) + oppiaPlayerService.getRandomSuffix());
+
+          $scope.upcomingStateName = null;
+          $scope.upcomingContentHtml = null;
+          $scope.upcomingInlineInteractionHtml = null;
+        }, TIME_FADEOUT_MSEC + 0.1 * TIME_HEIGHT_CHANGE_MSEC);
```

Comment:
I lean towards leaving this one as-is; it's fairly arbitrary. As long as it's before the transition finishes, its value shouldn't matter.